core/include/traccc/seeding/spacepoint_binning_helper.hpp

```diff
@@ -116,8 +116,8 @@ inline TRACCC_HOST_DEVICE size_t is_valid_sp(const seedfinder_config& config,
     if (spPhi > config.phiMax || spPhi < config.phiMin) {
         return detray::detail::invalid_value<size_t>();
     }
-    size_t r_index = getter::perp(
-        vector2{sp.x() - config.beamPos[0], sp.y() - config.beamPos[1]});
+    size_t r_index = static_cast<size_t>(getter::perp(
+        vector2{sp.x() - config.beamPos[0], sp.y() - config.beamPos[1]}));
```

Comment:
Casting this directly to an integer seems wrong, unless the bin size is exactly a unit of length 1... Are you sure this is correct?

Reply:
I was just fixing the warning:
`implicit conversion turns floating-point number into integer: 'float' to 'size_t' (aka 'unsigned long') [-Werror,-Wfloat-conversion] 
size_t r_index = getter::perp(`

Reply:
I'm suspicious of this line... I would have expected a division by the size of each r-bin in order to get an actual index. Again, this code only works if the bin size is exactly one. But I can open an issue regarding this.